scrapy/core/downloader/handlers/http11.py

```diff
@@ -31,8 +32,16 @@ def __init__(self, settings):
         self._pool = HTTPConnectionPool(reactor, persistent=True)
         self._pool.maxPersistentPerHost = settings.getint('CONCURRENT_REQUESTS_PER_DOMAIN')
         self._pool._factory.noisy = False
+
+        self._sslMethod = openssl_methods[settings.get('DOWNLOADER_CLIENT_TLS_METHOD')]
         self._contextFactoryClass = load_object(settings['DOWNLOADER_CLIENTCONTEXTFACTORY'])
-        self._contextFactory = self._contextFactoryClass()
+        # try method-aware context factory
+        try:
+            self._contextFactory = self._contextFactoryClass(method=self._sslMethod)
+        except TypeError:
+            # use defaults
+            self._contextFactory = self._contextFactoryClass()
```

Comment:
@dangra , I added the warning in https://github.com/redapple/scrapy/commit/c9890d5f7662dc2673696ceb80870c5318746495, but I'm not really happy with how it displays:

```
2016-02-24 01:12:51 [scrapy] INFO: Crawled 0 pages (at 0 pages/min), scraped 0 items (at 0 items/min)
2016-02-24 01:12:51 [py.warnings] WARNING: /home/paul/src/scrapy/scrapy/core/downloader/handlers/http11.py:48: UserWarning: 
    You are using a context factory class that does not accept the `method` argument
    (type OpenSSL.SSL method, e.g. OpenSSL.SSL.SSLv23_METHOD).
    Please upgrade your context factory class to handle or ignore it.
  Please upgrade your context factory class to handle or ignore it.""")

```

I don't understand why `Please upgrade your context factory class to handle or ignore it.""")` is repeated.
Any ideas? Any suggestions on making it prettier?